server/models/services/ModelService.js

```diff
@@ -0,0 +1,216 @@
+ *       as: 'roles'
+ *       attributes: [ 'name', 'permissions' ]
+ *     }
+ *   ]
+ * })
+ *
+ * @example
+ * // returns { attribute: 'value', associationData: [ { associationAttribute: 'associationAttributeValue' } ] }
+ * return ModelService.getById(Model, id, [ 'attribute' ], [
+ *   {
+ *     model: AssociationModel,
+ *     as: 'associationData',
+ *     attributes: [ 'associationAttribute' ]
+ *   }
+ * ]
+ *
+ * @param {Model} model - Sequelize Model instance to query for
+ * @param {number | string} id - ID of the Sequelize Model to query for
+ * @param {any[]} attributes - attributes of the Sequelize Model to be returned
+ * @param {any[]} include - information on Sequelize Model relationships
+ * @returns {Promise<Model>} - Sequelize Model
+ */
+const getById = async (model, id, attributes = [], include = []) => {
+    if (!model) throw new Error('Model not defined'); // TODO: pass through custom APIError
+
+    // findByPk
+    return await model.findOne({
+        where: { id },
+        attributes,
+        include
+    });
+};
+
+/**
+ * @param {Model} model - Sequelize Model instance to query for
+ * @param {object} queryParams - values to be used to query Sequelize Model
+ * @param {any[]} attributes - attributes of the Sequelize Model to be returned
+ * @param {any[]} include - information on Sequelize Model relationships
+ * @returns {Promise<Model>} - Sequelize Model
+ */
+const getByQuery = async (
+    model,
+    queryParams,
+    attributes = [],
+    include = []
+) => {
+    if (!model) throw new Error('Model not defined'); // TODO: pass through custom APIError
+
+    return await model.findOne({
+        where: { ...queryParams },
+        attributes,
+        include
+    });
+};
+
+/**
+ * @link {https://sequelize.org/v5/manual/models-usage.html#-code-findandcountall--code----search-for-multiple-elements-in-the-database--returns-both-data-and-total-count}
+ * @param {Model} model - Sequelize Model instance to query for
+ * @param {object} filter - values to be used to search Sequelize Model
+ * @param {any[]} attributes - attributes of the Sequelize Model to be returned
+ * @param {any[]} include - information on Sequelize Model relationships
+ * @param {object} pagination - pagination options for query
+ * @param {number} [pagination.page=0] - page to be queried in the pagination result (affected by {@param pagination.enable})
+ * @param {number} [pagination.limit=10] - amount of results to be returned per page (affected by {@param pagination.enable})
+ * @param {string[][]} [pagination.order=[]] - expects a Sequelize structured input dataset for sorting the Sequelize Model results (NOT affected by {@param pagination.enable}). See {@link https://sequelize.org/v5/manual/querying.html#ordering} and {@example [ [ 'username', 'DESC' ], [..., ...], ... ]}
+ * @param {boolean} [pagination.enable=false] - use to enable pagination for a query result as well useful values. Data for all items matching query if not enabled
```

Comment:
I think this enabled property isn't needed - not including the pagination object is enough to know there won't be any pagination!